cmd/thanos/compact.go

```diff
@@ -149,6 +149,8 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application) {
 	webPrefixHeaderName := cmd.Flag("web.prefix-header", "Name of HTTP request header used for dynamic prefixing of UI links and redirects. This option is ignored if web.external-prefix argument is set. Security risk: enable this option only if a reverse proxy in front of thanos is resetting the header. The --web.prefix-header=X-Forwarded-Prefix option can be useful, for example, if Thanos UI is served via Traefik reverse proxy with PathPrefixStrip option enabled, which sends the stripped prefix value in X-Forwarded-Prefix header. This allows thanos UI to be served on a sub-path.").Default("").String()
 	label := cmd.Flag("bucket-web-label", "Prometheus label to use as timeline title in the bucket web UI").String()
 
+	retainCompactedBlocks := cmd.Flag("retain-compacted-blocks", "Do not remove old blocks even after they are safely compacted, but move them to another directory").Default("false").Bool()
```

Comment:
Cool, we were thinking about this, but actually we though about separate bucket :thinking: 
Thoughts?

Essentially backup mode